lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/BuildJsonHttpDeserializer.kt

```diff
@@ -19,26 +21,54 @@ private data class JsonRpcResponse<T>(
     val result: T? = null,
 
     @SerialName("error")
+    @Serializable(with = JsonRpcErrorPolymorphicSerializer::class)
     val error: JsonRpcError? = null,
 )
 
+internal sealed class JsonRpcError {
+    @SerialName("code")
+    abstract val code: Int
+
+    @SerialName("message")
+    abstract val message: String
+
+    @SerialName("data")
+    abstract val data: Any?
+}
+
+@Serializable(with = JsonRpcStandardErrorSerializer::class)
+internal data class JsonRpcStandardError(
+    @SerialName("code")
+    override val code: Int,
+
+    @SerialName("message")
+    override val message: String,
+
+    @SerialName("data")
+    override val data: String? = null,
+) : JsonRpcError()
+
 @Serializable
-private data class JsonRpcError(
+internal data class JsonRpcContractError(
```

Comment:
Theoretically, it could be as well accepted as StandardError, but then error.toString() would look differently:
```
40 : "Contract error" : {"revert_error": "Revert error"}
```